### ISSUE TITLE
[Priority] for IMethodInfoAdvice

### ISSUE THREAD

**Issue by wilhelmmedetz** [ArxOne/MrAdvice#145]
IMethodInfoAdvice.Advise(MethodInfoAdviceContext context) does not seem to be invoked in correspondence with the Priority attribute.

**Comment by picrap** [comment]
Could you be more specific? Is this random? What do you expect and what do you get instead?

**Comment by wilhelmmedetz** [comment]
I made a unit test.

```
using System;
using ArxOne.MrAdvice.Advice;
using ArxOne.MrAdvice.Annotation;
using NUnit.Framework;

namespace SimControl.Samples.CSharp.TestsOld
{
    [AttributeUsage(AttributeTargets.Method), Priority(Priority)]
    public class Advice1Attribute: Attribute, IMethodInfoAdvice
    {
        public void Advise(MethodInfoAdviceContext context)
        {
            if (MethodInfoAdviceTests.Expected != Priority)
                throw new InvalidOperationException();

            MethodInfoAdviceTests.Expected = Priority-1;
        }

        public const int Priority = 1;
    }

    [AttributeUsage(AttributeTargets.Method), Priority(Priority)]
    public class Advice2Attribute: Attribute, IMethodInfoAdvice
    {
        public void Advise(MethodInfoAdviceContext context)
        {
            if (MethodInfoAdviceTests.Expected != Priority)
                throw new InvalidOperationException();

            MethodInfoAdviceTests.Expected = Priority-1;
        }

        public const int Priority = 2;
    }

    class ClassA
    {
        [Advice1, Advice2]
        public static void Method() { }
    }

    class ClassB
    {
        [Advice2, Advice1]
        public static void Method() { }
    }

    [TestFixture]
    public class MethodInfoAdviceTests
    {
        [Test]
        public void MethodInfoAdviceClassA_MethodA()
        {
            Expected = 2;
            ClassA.Method();
            Assert.That(Expected, Is.Zero);
        }

        [Test]
        public void MethodInfoAdviceClassB_MethodB()
        {
            Expected = 2;
            ClassB.Method(); // fails
            Assert.That(Expected, Is.Zero);
        }

        public static int Expected;
    }
}
```

**Comment by picrap** [comment]
Sorry, I misread the title and confused `IMethodInfoAdvice` with `IAdvice`. Right, the priority wasn't used for `IMethodInfoAdvice` (I never considered implementing it because I never thought it could be useful).
This is fixed in release 2.8.8.
